### PR TITLE
Add CLI documentation

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Add CLI documentation set
- Create structure for docu based on the README.md

**Related issue(s)**
See also #279

### CODE REVIEW

File: docs/01-01-overview.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: Overview
+---
+
+Kyma CLI is a commandline tool which supports Kyma developers. It provides a set of commands and flags you can use to: 
+
+- Provision a cluster locally or on cloud providers, such as AWS or Azure, or use Gardener to set up and manage your clusters.
```
Comment (suleymanakbas91) [inline docs/01-01-overview.md:7]: AWS is not supported :)
```suggestion
- Provision a cluster locally or on cloud providers, such as GKE or Azure, or use Gardener to set up and manage your clusters.
```

File: docs/03-03-examples.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+The following examples show how to use Kyma CLI.
+
+- Install Kyma with Minikube on Mac:
+
+    ```bash
+    kyma provision minikube
+    kyma install
+    ```
+
+- Install Kyma with Minikube on Windows:
```
Comment (suleymanakbas91) [inline docs/03-03-examples.md:15]: I think we can merge this with the previous example. "Install Kyma with Minikube on Mac or Windows:"

Comment (bszwarc) [reply docs/03-03-examples.md:15]: True 👍

File: docs/03-03-examples.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+---
+
+The following examples show how to provision a cluster, install Kyma, and run the tests.
+
+## Provision a cluster locally or using cloud providers
+
+To provision a cluster on a specific cloud provider (in this example GCP), run:
+
+```bash
+kyma provision gke -c {SERVICE_ACCOUNT_KEY_FILE_PATH} -n $CLUSTER_NAME -p $GCP_PROJECT
```
Comment (suleymanakbas91) [inline docs/03-03-examples.md:13]:
```suggestion
kyma provision gke -c {SERVICE_ACCOUNT_KEY_FILE_PATH} -n {CLUSTER_NAME} -p {GCP_PROJECT} 
```

File: docs/03-02-use-kyma-cli.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+kyma {COMMAND} {FLAGS}
+```
+
+- **{COMMAND}** specifies the operation you want to perform, such as provisioning the cluster or installing Kyma,
+- **{FLAGS}** specifies optional flags you can use to enrich you command.
+
+See the example:
+
+```bash
+kyma install --source=latest
```
Comment (suleymanakbas91) [inline docs/03-02-use-kyma-cli.md:20]: I think this looks better :)
```suggestion
kyma install -s latest
```

File: README.md [hunk 5]
```diff
@@ -92,76 +45,16 @@ Example:
 kyma install --source=latest
```
Comment (suleymanakbas91) [inline README.md:45]:
```suggestion
kyma install -s latest
```

File: docs/03-01-installation.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+### Chocolatey (Windows)
+
+To install Kyma CLI on Windows using [Chocolatey](https://www.chocolatey.org), run:
+
+```PowerShell
+choco install kyma-cli
+```
+You don't have to bump Kyma CLI Chocolatey package manually with each new release, as it includes a script that automatically checks for new releases and updates the package to the latest one.
+
+Still, the package requires some maintenance to keep its dedicated [site](https://chocolatey.org/packages/kyma-cli) at`chocolatey.org` up to datw. This means, you should regularly update the description, details, screenshots, etc. To keep the site up to date, submit a pull request to [Chocolatey's GitHub repository](https://github.com/dgalbraith/chocolatey-packages/tree/master/automatic/kyma-cli).
```
Comment (klaudiagrz) [inline docs/03-01-installation.md:67]:
```suggestion
Still, the package requires some maintenance to keep its dedicated [site](https://chocolatey.org/packages/kyma-cli) at`chocolatey.org` up to date. This means you should regularly update the description, details, screenshots, etc. To keep the site up to date, submit a pull request to [Chocolatey's GitHub repository](https://github.com/dgalbraith/chocolatey-packages/tree/master/automatic/kyma-cli).
```

File: docs/09-01-kubectl-plugin-tutorial.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+---
+title: Use Kyma CLI as kubectl plugin
+type: Tutorials
```
Comment (klaudiagrz) [inline docs/09-01-kubectl-plugin-tutorial.md:3]: Please change the file numeration to 08 as this number corresponds to the tutorial type.

File: docs/03-02-use-kyma-cli.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+---
+title: Use Kyma CLI
+type: Details
+---
+
+Kyma CLI comes with a set of commands, each of which has its own specific set of flags. Use these to provision the cluster locally or using a chosen cloud provider, install, and test Kyma.
```
Comment (klaudiagrz) [inline docs/03-02-use-kyma-cli.md:6]:
```suggestion
Kyma CLI comes with a set of commands, each of which has its own specific set of flags. Use them to provision the cluster locally or using a chosen cloud provider, install, and test Kyma. 
```

File: docs/03-02-use-kyma-cli.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+Kyma CLI comes with a set of commands, each of which has its own specific set of flags. Use these to provision the cluster locally or using a chosen cloud provider, install, and test Kyma. 
+
+For the commands and flags to work, they need to follow this syntax:
+
+```bash
+kyma {COMMAND} {FLAGS}
+```
+
+- **{COMMAND}** specifies the operation you want to perform, such as provisioning the cluster or installing Kyma
```
Comment (klaudiagrz) [inline docs/03-02-use-kyma-cli.md:14]:
```suggestion
- **{COMMAND}** specifies the operation you want to perform, such as provisioning the cluster or installing Kyma.
```

File: README.md [hunk 10]
```diff
@@ -2,80 +2,33 @@
...
-```bash
-curl -Lo kyma.tar.gz https://github.com/kyma-project/cli/releases/download/${KYMA_VERSION}/kyma_${ARCH}.tar.gz
-```
-
-## Usage
-
-### Syntax
-
-Use the following syntax to run the commands from your terminal:
+Once you have already installed the CLI, you can use its set of handy commands and flags to set up your cluster locally or on one of the cloud providers, install Kyma, manage and test it.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply README.md]: Yup, I did :)

File: README.md [hunk 11]
```diff
@@ -89,79 +42,19 @@ where:
...
-    ```
-
-- Install Kyma with Minikube on Windows using HyperV:
-
-    ```bash
-    kyma provision minikube --vm-driver hyperv --hypervVirtualSwitch {YOUR_SWITCH_NAME}
-    kyma install
-
-    ```
+>Read more about the available [commands and flags](https://github.com/kyma-project/cli/blob/master/docs/03-02-use-kyma-cli.md).
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply README.md]: Yes, agree - fixed it.

File: docs/09-01-kubectl-plugin-tutorial.md (hunk 7 above)
Comment (bszwarc) [reply docs/09-01-kubectl-plugin-tutorial.md]: Done :)

**Comment by bszwarc** [comment]
If there are any follow up things to do on this task, we will create a separate PR. I wanted to merge it today so that Maciej can work on the website part.

**Pull request merged**